tests/ui/type-alias-impl-trait/implied_lifetime_wf_check3.rs

```diff
@@ -5,7 +5,6 @@ mod test_lifetime_param {
     fn defining(a: &str) -> Ty<'_> { a }
     fn assert_static<'a: 'static>() {}
     fn test<'a>() where Ty<'a>: 'static { assert_static::<'a>() }
-    //~^ ERROR: lifetime may not live long enough
```

Comment:
:eyes:

Reply:
`WeakTy<'co>: 'static` now implies 'co: 'static` similar to ADTs. Is that fine / sound?

Reply:
it is, but we need tests that show that we actually enforce it, too.